### PR TITLE
Implement a HubSpot API client

### PR DESCRIPTION
**Pull request by alyakimenko**
### Description

This PR seeks to implement a basic HubSpot API client that can retrieve lists of resources.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-hubspot/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: hubspot/hubspot.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+// JSON decoded and stored in the value pointed to by out, or returned as an
+// error if an API error has occurred.
+func (c *Client) do(req *http.Request, out any) error {
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("http client do: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
```
Comment (maksenius) [inline hubspot/hubspot.go:90]: What about retry logic?

File: hubspot/hubspot_test.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+)
+
+// setup sets up a test HTTP server along with a hubspot.Client that is
+// configured to talk to that test server.
+func setup() (client *Client, mux *http.ServeMux, teardown func()) {
+	mux = http.NewServeMux()
+	server := httptest.NewServer(mux)
+
+	client = NewClient("secret", server.Client())
+	url, _ := url.Parse(server.URL)
```
Comment (maksenius) [inline hubspot/hubspot_test.go:38]: Could you please tell me why do we need do it?

Comment (alyakimenko) [reply hubspot/hubspot_test.go:38]: The `server.URL` is pointed to a test server that we start and can use within the unit tests for emulating the HubSpot API. It allows us to redefine the `defaultBaseURL` which is equal to `https://api.hubapi.com`.

File: hubspot/hubspot.go (hunk 1 above)
Comment (alyakimenko) [reply hubspot/hubspot.go:90]: I'll add it later when I'll implement the Source

File: hubspot/hubspot.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		return &UnexpectedStatusCodeError{
```
[the remark this answers is not in the document]
Comment (alyakimenko) [reply hubspot/hubspot.go]: Good point! I've added a body as `[]byte` to the `UnexpectedStatusCodeError` struct.

**Review by maksenius — APPROVED** [review]

**Pull request merged**